### DISCUSSION TITLE
Building Multiple Binaries with a Single Command

### DISCUSSION THREAD

**Discussion by ryancurrah** [goreleaser/goreleaser#4886] [Q&A] [answered]
Hello GoReleaser Team,

I've been exploring various methods to optimize the build process. I came across an approach (https://blog.howardjohn.info/posts/go-build-times/#building-multiple-binaries) that significantly reduces build time by compiling multiple binaries in a single `go build` command, as demonstrated below:

```bash
go build -o /tmp/ $BINARIES
```

Where `$BINARIES` includes paths to various main packages.

This method proved to be faster compared to building binaries sequentially or even in parallel. I am curious to know if GoReleaser supports or can be configured to utilize this approach to build multiple binaries in a single command to leverage concurrent builds and caching more effectively. Is this possible with GoReleaser, or are there any workarounds you would recommend?

Thank you for your assistance!

**Comment by caarlos0** [comment]
hey, goreleaser cannot do this right now I think... but looks like something that can be added as a feature.

I'll look into it for v2.1

**Reply by ryancurrah** [reply]
Ok cool. I was giving it some more thought last night, I think it would not be possible when the options for each build are different. Only when they are the same, I think.

**Reply by caarlos0** [reply]
yes, very likely

**Comment by caarlos0** [comment, marked as answer]
being tracked in https://github.com/goreleaser/goreleaser/issues/4927

will close this one
